ACCS-365: Payment Services: Stored Cards at Checkout (#821)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/payment-services/slots.mdx

---

import TableWrapper from '@components/TableWrapper.astro';

The Payment Services drop-in does not expose any slots for customization.

<TableWrapper nowrap={[0]}>

| Container | Slots |
| --- | --- |
| ApplePay | None (SDK-rendered) |
| CreditCard | None (SDK-rendered) |
| StoredCards | None (optional vaulted-card UI; use props, dictionary, and styles) |

</TableWrapper>

## Why no slots?

This drop-in wraps the Adobe Payment Services SDK (`@adobe-commerce/payment-services-sdk`), which renders secure payment forms directly into specified DOM elements. The SDK controls all UI rendering to maintain PCI (Payment Card Industry) compliance and security standards. You customize the payment forms through SDK configuration options (field placeholders, card type settings, callback handlers) passed to `sdk.Payment.CreditCard.render()`, not through the slot-based pattern other drop-ins use.

> **StoredCards** (optional, for vaulted saved cards at checkout) is separate from the SDK card form and also does not define slots. When you adopt it, customize layout and copy through [container props](/dropins/payment-services/containers/stored-cards/#configuration), the [dictionary](/dropins/payment-services/dictionary/), and [styles](/dropins/payment-services/styles/).

Reviewer: ```suggestion
`StoredCards` (optional, for vaulted cards at checkout) is separate from the SDK card form and does not provide slots. When used, customize the layout and copy through [container props](/dropins/payment-services/containers/stored-cards/#configuration), the [dictionary](/dropins/payment-services/dictionary/), and [styles](/dropins/payment-services/styles/).
```
